### ISSUE TITLE
NovoerAdapter and XTandem unexpected parse error 

### ISSUE THREAD

**Issue by giangiacomoberetta1** [OpenMS/OpenMS#4674]
Hi !

While processing a big mzML file (dimension around 60 Gb), for XTandem and NovorAdapter I got the following error message, while MSGF+ and SpectraSTAdapter finished smoothly their search.
There are some clues that the error is generated by some memory usage settings, but I am unsure about it...

Additional problem (which I guess to be related to the one above), the subsequent PIA fails in stackoverflow error.

I will be grateful for help, as I would like to fix these problems.

Thanks in advance !

Error: Unable to read file (- due to that error of type Parse Error in: C:\jenkins\ws\openms_release_packaging\9447518b\source\src\openms\source\FORMAT\HANDLERS\MzMLHandler.cpp@176-void __cdecl OpenMS::Internal::MzMLHandler::populateSpectraWithData_(void))

**Comment by jpfeuffer** [comment]
Memory issues may indeed be the cause. How much RAM do you have available?
While we can reduce the RAM needed inside the search engine adapters quickly, this will be harder for other tools (e.g. FeatureFinders). With PIA you might have hit another bottleneck already (depends on the number of IDs you get out of the searches).

**Comment by giangiacomoberetta1** [comment]
Hi!

As I have 16 Gb installed RAM. KNIME was running with 8 Gb RAM available, I increased it to all available through knime.ini file, without improvement.

In this case FF is a minor issue as I am interested in peptide identifications and protein inference only.

Regarding found IDs, PeptideIndexer is producing around 120.000 IDs.

**Comment by jpfeuffer** [comment]
Oh.. I guess I have bad news then, since we currently double-check centroidedness of the spectra it loads all MS2 spectra first in NovoR and XTandem Adapter. We will fix that but until then I think there is no way that it will fit into the remaining 8GB.
120.000 is according to my experience also too much for 8GB RAM and PIA. You will need to do heavy pre-filtering, either with FDR+IDFilter or in the PIA settings.

**Comment by giangiacomoberetta1** [comment]
OK, I got it, thanks!

Currently I cannot access my 64 Gb RAM pc, due to CV-19 restrictions.
I will try on it too and keep you posted.

Thanks again!

**Comment by jpfeuffer** [comment]
Cool. See #4675 for the progress on the adapters' memory usage.

**Comment by jpfeuffer** [comment]
By the way, could you please double-check your centroiding?
E.g. what tool did you use? Which command? The file size and the number of peaks in your spectra (see  separate Novor issue) might indicate that the data was not correctly centroided.

**Comment by giangiacomoberetta1** [comment]
Yes of course, centroiding was done with Proteowizard (peak peaking option)

**Comment by giangiacomoberetta1** [comment]
Meanwhile I tried inference with FidoAdapter anf MSG+ for peptide identification and its working
